src/main/java/amazin/model/Book.java

```diff
@@ -147,12 +179,25 @@ public Book(Long id, String name, String description, String ISBN, String pictur
         this.publisher = publisher;
         this.inventory = inventory;
         this.price = price;
+        this.tags = tags;
     }
 
     @Override
     public String toString() {
-        return "Book{" + "id=" + id + ", name='" + name + '\'' + ", description='" + description + '\'' + ", ISBN="
-                + ISBN + ", picture='" + picture + '\'' + ", author='" + author + '\'' + ", publisher='" + publisher
+        String returnString = "Book{" + "id=" + id + ", name='" + name + '\'' + ", description='"
+                               + description + '\'' + ", tags='[";
```

Comment:
Why don't you add the "tags" string in an if statement about there being tags, rather than removing part of the string it tags don't exist at the end?

Reply:
Tags is an empty list when there is no tags. I remove the last comma not the tags string, and it is the neatest way to do it.